src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t connect to DB", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement
+                     = connection.prepareStatement(query)) {
+            statement.setObject(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = getBookFromResultSet(resultSet);
+                return Optional.of(book);
+            }
+            throw new DataProcessingException("Can`t find book by id " + id);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t connect to DB", e);
+        }
+
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement
+                = connection.prepareStatement(query)) {
+            try (ResultSet resultSet = statement.executeQuery()) {
+                while (resultSet.next()) {
+                    books.add(getBookFromResultSet(resultSet));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get books from the DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement
+                = connection.prepareStatement(query)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setObject(3, book.getId());
+            int updatedRows = statement.executeUpdate();
+            if (updatedRows < 1) {
+                throw new DataProcessingException("Updating book by id "
+                        + book.getId() + " failed.");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t connect to DB", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement
+                = connection.prepareStatement(query)) {
+            statement.setObject(1, id);
+            int updatedRows = statement.executeUpdate();
+            return updatedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t connect to DB", e);
```

Comment:
Too general error message + if we got SQLException it is not always Can not connect to db